### DISCUSSION TITLE
error while starting ltsp image

### DISCUSSION THREAD

**Discussion by jorigel** [ltsp/ltsp#427] [Q&A]
Hi,
while starting update of the image via
`ltsp image [imagename]`
I get the error message:

```
LTSP command failed: test MAC_ADDRESS= != MAC_ADDRESS=
Aborting ltsp
```
What can I do about it?
Thank You

**Comment by alkisg** [comment]
Hi jorigel, this message comes from 55-config.sh, function network_vars().

What's the output of these commands in your server?

```shell
ip r
ip -o route get 192.168.67.1
ip -o route get 192.168.67.2
```

**Comment by jorigel** [comment]
Hi,
it says:

```
# ip r
0.0.0.0/1 via 10.8.0.1 dev tun0 
default via 10.1.1.200 dev bond0 
10.1.1.0/24 dev bond0 proto kernel scope link src 10.1.1.204 
10.8.0.0/24 dev tun0 proto kernel scope link src 10.8.0.4 
128.0.0.0/1 via 10.8.0.1 dev tun0 
169.254.0.0/16 dev bond0 scope link metric 1000 
213.152.188.3 via 10.1.1.200 dev bond0 
 
# ip -o route get 192.168.67.1
192.168.67.1 via 10.8.0.1 dev tun0 src 10.8.0.4 uid 0 \    cache
# ip -o route get 192.168.67.2
192.168.67.2 via 10.8.0.1 dev tun0 src 10.8.0.4 uid 0 \    cache 

```
10.8.0.1 - tun0 device to the internet
10.1.1.200 - router
10.1.1.204 - bond0 bonded device of the server

Thank You, hope that helps
regards
jori

**Comment by alkisg** [comment]
jori what is tun0, OpenVPN? It's claiming to be the default route for everything...

So LTSP can't detect which network card to use for 192.168.67.x.

Try adding a manual route via bond0 with something like the following:

```shell
ip route add 192.168.67.0/24 via 10.1.1.200 dev bond0
```

**Comment by jorigel** [comment]
Hi Alkis,
indeed this works. What You asked for, alkis, yes it is. 
May I ask another stupid question? Where is that net (192.168.67.0/24) coming from and why does it need a direct route to the native gateway? Is it  not able to use the tunneled one into the internet? I thought dnsmask is looking for my real bindery and dhcp-management, but why then routing to the native gateway?
I know, stupid me, all these years and still learning :) I just programmed a smart home management steering within others a pellet heating all controlled by the excess energy of a photo-voltaic system and still failing on issues like this :(

Thank You my dear, as always: You are the best

regards jori

**Comment by alkisg** [comment]
One very common LTSP setup is the "2 NIC setup". In that, the internal NIC is assigned the IP of 192.168.67.1, and dnsmasq is automatically configured to be a real DHCP server in that subnet.

So when LTSP wants to detect the "client or server MAC_ADDRESS", it tries to detect the one that talks to this subnet first, so that the traffic goes through LAN and not via the Internet gateway.
